RecoEgamma/PhotonIdentification/src/PhotonMVABasedHaloTagger.cc

```diff
@@ -0,0 +1,445 @@
+
+  double angle_EE_HE_samedR =
+      calAngleBetweenEEAndSubDet(hcalClusNhits_samedR_, hcalClusX_samedR_, hcalClusY_samedR_, hcalClusZ_samedR_);
+
+  double angle_EE_ES_samedPhi = calAngleBetweenEEAndSubDet(
+      preshowerNhits_samedPhi_, preshowerX_samedPhi_, preshowerY_samedPhi_, preshowerZ_samedPhi_);
+
+  double angle_EE_ES_samedR =
+      calAngleBetweenEEAndSubDet(preshowerNhits_samedR_, preshowerX_samedR_, preshowerY_samedR_, preshowerZ_samedR_);
+
+  ////set all the above calculated variables as input to the MVA
+
+  const auto& vCov = lazyToolnoZS.localCovariances(*(pho->superCluster()->seed()));
+  double spp = (isnan(vCov[2]) ? 0. : sqrt(vCov[2]));
+
+  ///https://cmssdt.cern.ch/lxr/source/RecoEgamma/ElectronIdentification/src/ElectronMVAEstimator.cc
+
+  float vars[15];
+
+  vars[0] = preshowerE_samedPhi_;
+  vars[1] = hcalClusE_samedPhi_;
+  vars[2] = preshowerE_samedR_;
+  vars[3] = hcalClusE_samedR_;
+  vars[4] = pho->full5x5_r9();
+  vars[5] = pho->superCluster()->etaWidth();
+  vars[6] = pho->superCluster()->phiWidth();
+  vars[7] = pho->full5x5_sigmaIetaIeta();
+  vars[8] = spp;
+  vars[9] = angle_EE_ES_samedR;
+  vars[10] = angle_EE_HE_samedR;
+  vars[11] = angle_EE_ES_samedPhi;
+  vars[12] = angle_EE_HE_samedPhi;
+  vars[13] = (pho->superCluster()->preshowerEnergyPlane1() + pho->superCluster()->preshowerEnergyPlane2()) /
+             pho->superCluster()->rawEnergy();
+  vars[14] = rho_;
+
+  double BHmva = gbr_->GetAdaBoostClassifier(vars);
+  return BHmva;
+}
+
+void PhotonMVABasedHaloTagger::calphoClusCoordinECAL(const CaloGeometry* geo,
+                                                     const reco::Photon* pho,
+                                                     const EcalPFRecHitThresholds* thresholds,
+                                                     edm::Handle<EcalRecHitCollection> ecalrechitCollHandle) {
+  ecalClusX_ = 0;
+  ecalClusY_ = 0;
+  ecalClusZ_ = 0;
+  ecalClusNhits_ = 0;
+  ecalClusE_ = 0;
+
+  double phoSCEta = pho->superCluster()->eta();
+  double phoSCPhi = pho->superCluster()->phi();
+
+  const EcalRecHitCollection* ecalRecHits = ecalrechitCollHandle.product();
+
+  for (EcalRecHitCollection::const_iterator ecalrechit = ecalRecHits->begin(); ecalrechit != ecalRecHits->end();
+       ecalrechit++) {
+    auto const det = ecalrechit->id();
+    double rhE = ecalrechit->energy();
+    const GlobalPoint& rechitPoint = geo->getPosition(det);
+
+    double rhEta = rechitPoint.eta();
+    double rhPhi = rechitPoint.phi();
+    double rhX = rechitPoint.x();
+    double rhY = rechitPoint.y();
+    double rhZ = rechitPoint.z();
+
+    if (thresholds == nullptr) {
+      throw cms::Exception("EmptyPFRechHitThresCollection")
+          << "In PhotonMVABasedHaloTagger::calphoClusCoordinECAL, EcalPFRecHitThresholds cannot be = nulptr";
+    }
+
+    float rhThres = 0.0;
+    if (thresholds != nullptr) {
```

Comment:
this check should not be required, as you throw an exception for nullptr a few lines above